lib/ui/text/paragraph.h

```diff
@@ -50,6 +50,7 @@ class Paragraph : public RefCountedDartWrappable<Paragraph> {
   std::vector<TextBox> getRectsForPlaceholders();
   Dart_Handle getPositionForOffset(double dx, double dy);
   Dart_Handle getWordBoundary(unsigned offset);
+  std::vector<LineMetrics> getLineMetrics();
```

Comment:
This looks like an expensive data structure, because it doesn't use pointers. Passing it around would involve, copying the vector, copying each `LineMetrics` object, and recursively copying `std::map<size_t, txt::RunMetrics>` that's inside it.